bot/cogs/filtering.py

```diff
@@ -368,6 +399,46 @@ def mod_log(self) -> ModLog:
         except discord.errors.Forbidden:
             await channel.send(f"{filtered_member.mention} {reason}")
 
+    async def _scheduled_task(self, msg: dict) -> None:
+        """A coroutine which delete the offensive message once the delete date is reached."""
+        delete_at = datetime.datetime.fromisoformat(msg['delete_date'][:-1])
```

Comment:
It'd be better to do this with dateutil as done [here](https://github.com/python-discord/bot/blob/master/bot/cogs/moderation/scheduler.py#L410).